master/src/main/scala/org/apache/celeborn/service/deploy/master/tags/ql/TagsQLParser.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.service.deploy.master.tags.ql
+
+sealed trait Operator
+case object Equals extends Operator
+case object NotEquals extends Operator
+
+case class Node(key: String, operator: Operator, values: Set[String])
+
+/**
+ * TagsQL uses key/value pair to give your tags more context.
+ *
+ * The query language supports the following syntax:
```

Comment:
In what scenarios might you need this functionality? 

Can the following rules align with your goal?
`tag1,tag2 | tag3,tag4 | tag5,tag6`

In the expression above, `tag1` and `tag2` are combined using a logical AND, just like `tag3` and `tag4`. The results of `tag1` AND `tag2` and `tag3` AND `tag4` are combined using a logical OR.